File: ide_copy/pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //card no and copies for operations
    long n, copy = 0, copy1; 
    int len = 0, sum = 0, flag = 0, i;
    string type = "";
    
    //input
    do
    {
        n = get_long("Number: ");
    }
    while (n <= 0);
    copy1 = n;
    
    //reverse
    while (n != 0)
    {
        copy = (copy * 10) + (n % 10);
        n = n / 10;
        len += 1;
    }
    
    //preliminary test
    if ((copy % 100 == 43 || copy % 100 == 73) && len == 15)
    {
        type = "AMEX";
    }
    else if ((copy % 100 == 15 || copy % 100 == 25 || copy % 100 == 35 || copy % 10 == 45 || copy % 100 == 55) && len == 16)
    {
        type = "MASTERCARD";
    }
    else if (copy % 10 == 4 && (len == 13 || len == 16))
    {
        type = "VISA";
    }
    else
    {
        //Incase all test cases fail 
        flag = 1;
    }
    
    //Luhn algorithm implementation
    while (copy1 != 0)
    {
        //adding non doubled elements to sum
        sum += (copy1 % 10);
        copy1 /= 10;
        
        //doubling every other digit
        i = (copy1 % 10) * 2;
        while (i != 0)
        {
            //splitting douled elements
            sum += (i % 10);
            i /= 10;
        }
        copy1 /= 10;
    }
    
    //final check for compatibility
    if (sum % 10 == 0 && flag != 1)
    {
        printf("%s\n", type);
    }
    else
    {
        printf("INVALID\n");
    }
    
}

File: ide_copy/pset6/credit/credit.py
from cs50 import *

# Translation from pset1
# Card no. input
while True:
    n = get_int("Number: ")
    if n > 0:
        break

# Initializing all temporary values    
copy, copy1, l, flag = 0, n, 0, 0

# Reversing
while n != 0:
    copy = (copy * 10) + (n % 10)
    n = n // 10
    # Computing length as l
    l += 1

# Preliminary test
if copy % 100 in [43, 73] and l == 15:
    name = "AMEX"
elif copy % 100 in [15, 25, 35, 45, 55] and l == 16:
    name = "MASTERCARD"
elif copy % 10 == 4 and (l == 13 or l == 16):
    name = "VISA"
else:
    # Incase all test cases fail
    flag = 1

# Initializing s to 0
s = 0

# Luhn algorithm implementation
while (copy1 != 0):

    # Adding non doubled elements to s
    s += (copy1 % 10)
    copy1 //= 10

    # Doubling every other digit
    i = (copy1 % 10) * 2
    while (i != 0):
        # Splitting doubled elements
        s += (i % 10)
        i //= 10
    copy1 //= 10

# Final check for compatibility
if (s % 10 == 0 and flag != 1):
    print(f"{name}")
else:
    print("INVALID")
